File: DTO/Category/Program/Dtos/ProgramDto.cs
using System.ComponentModel.DataAnnotations;
using DTO.Base;
using FluentValidation;

namespace DTO.Category.Program.Dtos;

public class ProgramDto : DtoBase
{
    [Required(AllowEmptyStrings = false, ErrorMessage = "Name is required")]
    public string? Name { get; set; }
}

public class ProgramDtoValidator : AbstractValidator<ProgramDto>
{
    public ProgramDtoValidator()
    {
        RuleFor(r => r.Name).NotEmpty().WithMessage("Name is not null");
    }
}

File: CLIENT/src/app/models/category/program/dtos/programDto.ts
import {createDefaultDtoBase, DtoBase} from "../../../base/dtoBase";

export interface ProgramDto extends DtoBase {
  name: string | null;
  note: string | null;
}

export function createDefaultProgramDto(): ProgramDto {
  let dtoBase: DtoBase = createDefaultDtoBase();
  return {
    name: null,
    note: null,
    ...dtoBase
  }
}
